custom_components/candy/client/model.py

```diff
@@ -181,28 +185,279 @@ class OvenState(StatusCode):
+    IDLE = (0, "Idle")
+    FAN = (1, "Fan")
+    LIGHT = (2, "Light")
+    FAN_LIGHT = (3, "Fan & Light")
+
+
+@dataclass
+class HoodStatus:
+    machine_state: HoodState
+    light: bool
+    fan: bool
+    grease_filter: bool
+    carbon_Filter: bool
+    warning: bool
+    remote_control: bool
+
+    @classmethod
+    def from_json(cls, json):
+        return cls(
+            machine_state=HoodStatus.parse_state(json),
+            light=json["Light"] == "1",
+            fan=int(json["Fan"]) > 0,
+            grease_filter=json["GreaseFilter"] == "1",
+            carbon_Filter=json["CarbonFilter"] == "1",
+            warning=json["Warning"] == "1",
+            remote_control=json["WiFiStatus"] == "1",
+        )
+
+    @staticmethod
+    def parse_state(json) -> str:
+        """
+        Parse Hood status
+        """
+        if int(json["Fan"]) > 0 and json["Light"] == "1":
+            return HoodState.FAN_LIGHT
+
+        if int(json["Fan"]) > 0:
+            return HoodState.FAN
+
+        if json["Light"] == "1":
+            return HoodState.LIGHT
+        else:
+            return HoodState.IDLE
+
+
+class FridgeState(StatusCode):
+    NORMAL = (0, "Normal")
+    ECO = (1, "Eco")
+    SUPER_FREEZING = (2, "Super freezing")
+    SMART_COOLING = (3, "Smart cooling")
+    SUPER_FREEZING_COOLING = (4, "Super freezing & cooling")
+    ERROR = (5, "Error")
+
+
+@dataclass
+class FridgeStatus:
+    machine_state: HobState
+    cooling_temperature: bool
+    freezing_temperature: bool
+    eco_mode: bool
+    super_freezing_mode: bool
+    smart_cooling_mode: bool
+    door_locked: bool
+    door_open: bool
+    fan: int
+    error: Optional[int]
+    remote_control: bool
+
+    @classmethod
+    def from_json(cls, json):
+        return cls(
+            machine_state=FridgeStatus.parse_state(json),
+            cooling_temperature=int(json["FrSet"]),
+            freezing_temperature=int(json["FzSet"]),
+            eco_mode=json["Eco"] == "1",
+            super_freezing_mode=json["Spr"] == "1",
+            smart_cooling_mode=json["Ice"] == "1",
+            door_locked=json["Lck"] == "1",
+            door_open=json["Door"] == "1",
+            fan=int(json["Fan"]),
+            error=int(json["Error"]) if int(json["Error"]) != 0 in json else None,
+            remote_control=json["WiFi"] == "1",
+        )
+
+    @staticmethod
+    def parse_state(json) -> str:
+        """
+        Parse Fridge status
+        """
+
+        if json["Error"] != "0":
+            return FridgeState.ERROR
+
+        if json["Eco"] == "1":
+            return FridgeState.ECO
+
+        if json["Ice"] == "1" and json["Spr"] == "1":
+            return FridgeState.SUPER_FREEZING_COOLING
+
+        if json["Ice"] == "1":
+            return FridgeState.SUPER_FREEZING
+
+        if json["Spr"] == "1":
+            return FridgeState.SMART_COOLING
+        else:
+            return FridgeState.NORMAL
+
+
+def only_numerics(seq):
```

Comment:
Can you add a comment explaining why this is needed and what it's doing? It's not clear for me what is `seq` and what is the output of this function.

Reply:
This method parse working status of fridge.  We have option errr, eco, super freezing and super coolind, super freezing, smart cooling and normal working.  This is depend od Fridge settings.